PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/IDEHelperImpl.java

```diff
@@ -393,6 +409,69 @@ public void openAppServiceFile(final AppServiceFile file, Object context) {
+        final FileEditor[] editors = fileEditorManager.openFile(virtualFile, true, true);
+        if (editors.length == 0) {
+            return false;
+        }
+        for (FileEditor fileEditor : editors) {
+            if (fileEditor instanceof TextEditor) {
+                final String originContent = getTextEditorContent((TextEditor) fileEditor);
+                final MessageBusConnection messageBusConnection = fileEditorManager.getProject().getMessageBus().connect(fileEditor);
+                messageBusConnection.subscribe(FileEditorManagerListener.Before.FILE_EDITOR_MANAGER, new FileEditorManagerListener.Before() {
+                    @Override
+                    public void beforeFileClosed(FileEditorManager source, VirtualFile file) {
+                        try {
+                            final String content = getTextEditorContent((TextEditor) fileEditor);
+                            if (file == virtualFile && !StringUtils.equals(content, originContent)) {
+                                boolean result = DefaultLoader.getUIHelper().showYesNoDialog(
+                                    fileEditor.getComponent(), SAVE_CHANGES, APP_SERVICE_FILE_EDITING, Messages.getQuestionIcon());
+                                if (result) {
+                                    saveFileToAzure(appServiceFile, content, fileEditorManager.getProject());
+                                }
+                            }
+                        } catch (RuntimeException e) {
+                            AzureExceptionHandler.getInstance().handleException(e);
+                        } finally {
+                            messageBusConnection.disconnect();
+                        }
+                    }
+                });
+            }
+        }
+        return true;
+    }
+
+    private static String getTextEditorContent(TextEditor textEditor) {
+        return textEditor.getEditor().getDocument().getText();
+    }
+
+    @AzureOperation(
+        value = "save file[%s] to azure",
+        params = {"appServiceFile.getName()"},
+        type = AzureOperation.Type.SERVICE
+    )
+    private void saveFileToAzure(final AppServiceFile appServiceFile, final String content, final Project project) {
+        AzureTaskManager.getInstance().runInBackground(new AzureTask(project, String.format("Saving %s", appServiceFile.getName()), false, () -> {
+            final AppServiceFileService fileService = AppServiceFileService.forApp(appServiceFile.getApp());
+            final AppServiceFile target = fileService.getFileByPath(appServiceFile.getPath());
+            if (target == null) {
+                boolean result = DefaultLoader.getUIHelper().showYesNoDialog(null, String.format(FILE_HAS_BEEN_DELETED, appServiceFile.getName()),
+                                                                             APP_SERVICE_FILE_EDITING, Messages.getQuestionIcon());
+                if (!result) {
```

Comment:
combine the if clause with previous showYesNoDialog

Reply:
`if(DefaultLoader.getUIHelper().showYesNoDialog(
                    null, String.format(FILE_HAS_BEEN_MODIFIED, appServiceFile.getName()), APP_SERVICE_FILE_EDITING, Messages.getQuestionIcon()))`? I think the if condition may be too long to understand